src/App.tsx

```diff
@@ -15,48 +17,111 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  alphabet = 'alphabet',
+  length = 'length',
+  reverse = 'reverse',
+}
+
+interface SomeParams{
+  sortText: string;
+}
+
+function getReadyGoods(goods: string[], { sortText }: SomeParams) {
+  const readyGoods = [...goods];
+
+  if (sortText) {
+    readyGoods.sort((good1, good2) => {
+      switch (sortText) {
+        case SortType.alphabet:
+          return good1.localeCompare(good2);
+
+        case SortType.length:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return readyGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortText, setSortText] = useState('');
+  const [reverseText, setReverseText] = useState('');
+  let sortedGoods = getReadyGoods(goodsFromServer, { sortText });
+
+  if (sortText !== SortType.reverse && reverseText === SortType.reverse) {
+    sortedGoods = sortedGoods.reverse();
+  } else if (reverseText === SortType.reverse) {
+    sortedGoods = goodsFromServer.reverse();
+  } else if (sortText === SortType.alphabet
+    && reverseText === SortType.reverse) {
+    sortedGoods = getReadyGoods(sortedGoods, { sortText });
+  }
   return (
     <div className="section content">
       <div className="buttons">
         <button
+          onClick={() => setSortText(SortType.alphabet)}
           type="button"
-          className="button is-info is-light"
+          className={cn('button', 'is-info', {
+            'is-light': sortText !== SortType.alphabet,
+          })}
         >
           Sort alphabetically
         </button>
 
         <button
+          onClick={() => setSortText(SortType.length)}
           type="button"
-          className="button is-success is-light"
+          className={cn('button', 'is-success', {
+            'is-light': sortText !== SortType.length,
+          })}
         >
           Sort by length
         </button>
 
         <button
+          onClick={reverseText ? () => {
+            setReverseText('');
+          } : () => {
+            setReverseText(SortType.reverse);
+          }
+          }
```

Comment:
Pls move the function inside `onClick` handler to a separate const and use as a callback here